GPL/EventProbe/EventProbe.bpf.c

```diff
@@ -29,22 +29,103 @@
 
 char LICENSE[] SEC("license") = "GPL";
 
-SEC("fexit/security_path_unlink")
-int BPF_PROG(security_path_unlink_exit, const struct path *dir, struct dentry *dentry, long ret)
+enum ebpf_fileevents_tid_state_id {
+    EBPF_FILEEVENTS_TID_STATE_UNKNOWN = 0,
+    EBPF_FILEEVENTS_TID_STATE_UNLINK  = 1,
+};
+
+struct ebpf_fileevents_unlink_state {
+    struct vfsmount *mnt;
+};
+struct ebpf_fileevents_tid_state {
+    enum ebpf_fileevents_tid_state_id state_id;
+    union {
+        struct ebpf_fileevents_unlink_state unlink;
+    } state;
+};
+
+struct bpf_map_def SEC("maps") elastic_ebpf_fileevents_tid_state = {
+    .type        = BPF_MAP_TYPE_LRU_HASH,
+    .key_size    = sizeof(u64),
+    .value_size  = sizeof(struct ebpf_fileevents_tid_state),
+    .max_entries = 4096,
+};
+
+static __always_inline struct ebpf_fileevents_tid_state *ebpf_fileevents_write_state__get(void)
 {
+    u64 tid = bpf_get_current_pid_tgid();
+    return bpf_map_lookup_elem(&elastic_ebpf_fileevents_tid_state, &tid);
+}
+
+static __always_inline long
+ebpf_fileevents_write_state__set(struct ebpf_fileevents_tid_state *state)
+{
+    u64 tid = bpf_get_current_pid_tgid();
+    return bpf_map_update_elem(&elastic_ebpf_fileevents_tid_state, &tid, state, BPF_ANY);
+}
+
+// todo(fntlnz): this is a temporary workaround, the userspace program should do this
+// just for reference.
+// 2 is on 5.14 on arch
+// 1 is on 5.10 on al2
+// 1 is on 5.10.68 on COS
+const volatile int fentry__vfs_unlink__dentry_parm = 0;
+
+SEC("fentry/do_unlinkat")
+int BPF_PROG(fexit__do_unlinkat)
+{
+    struct ebpf_fileevents_tid_state state;
+    __builtin_memset(&state, 0, sizeof(struct ebpf_fileevents_tid_state));
+    state.state_id = EBPF_FILEEVENTS_TID_STATE_UNLINK;
+    ebpf_fileevents_write_state__set(&state);
+
+    return 0;
+}
+
+SEC("fentry/mnt_want_write")
+int BPF_PROG(fexit__mnt_want_write)
```

Comment:
here we can use 
```suggestion
int BPF_PROG(fexit__mnt_want_write, struct vfsmount *m)
```
or you wanted to fill the map from userspace for all funcs/params even if nothing recently changed?

Reply:
you are right, I wasn't able to see this changing in the kernels I played with. We can probably just use parameter expansion here and access context directly after if we have th eneed

Reply:
done, thank you!